src/ripple/unl/tests/Sim3.h

```diff
@@ -0,0 +1,513 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012, 2013 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_UNL_SIM3_H_INCLUDED
+#define RIPPLE_UNL_SIM3_H_INCLUDED
+
+#include <ripple/unl/tests/BasicNetwork.h>
+#include <beast/container/aged_unordered_map.h>
+#include <boost/container/flat_set.hpp>
+#include <boost/optional.hpp>
+#include <algorithm>
+#include <memory>
+#include <sstream>
+#include <unordered_map>
+#include <vector>
+
+namespace ripple {
+namespace test {
+
+template <class Log>
+struct Sim3
+{
+    struct Config
+    {
+        int unl;
+        int peers = 100;
+        int trial = 100;
+    };
+
+    static int const nDegree = 10;      // outdegree
+    static int const nItem = 10;        // number of items
+    static int const nUpdateMS = 700;
+
+    using NodeKey = int;    // identifies a consensus participant
+    using ItemKey = int;    // identifies a ballot item
+    using ItemSet = boost::container::flat_set<ItemKey>;
+    using clock_type = std::chrono::system_clock;
+    using millis = std::chrono::milliseconds;
+
+    struct Network;
```

Comment:
`Network` is defined with a class below. Change either this decl to `class` of the other to `struct`